optuna/trial/_trial.py

```diff
@@ -768,7 +769,7 @@ def distributions(self):
             A dictionary containing all distributions.
         """
 
-        return self.storage.get_trial(self._trial_id).distributions
+        return copy.deepcopy(self.storage.get_trial(self._trial_id).distributions)
```

Comment:
Should the `user_attrs` and `system_attrs` custom getters perform copied in a similar manner to this custom getter for `distributions`?

Reply:
Yes, we need deep-copies in user_attrs and system_attrs, and params as well. Thank you for pointing them out.